runtime/onert/core/include/util/ObjectManager.h

```diff
@@ -36,33 +36,55 @@ namespace util
 template <typename Index, typename Object> class ObjectManager
 {
 public:
-  ObjectManager() : _index_count{0u} {}
+  ObjectManager() : _next_index{0u} {}
 
 public:
   /**
-   * @brief Create an object with args and put it in the container with a new Index for that
+   * @brief Create an object with args and put it in the container with a newly assigned @c Index
    *
    * @param[in] args Arguments for creating Operand object
-   * @return Created index that is associated to the object
+   * @return Created index that is associated to the object if successful, Undefined index otherwise
    */
   template <class... Args> Index emplace(Args &&... args)
   {
     auto index = generateIndex();
+    if (!index.valid())
+      return index;
     _objects.emplace(index, std::make_unique<Object>(std::forward<Args>(args)...));
     return index;
   }
 
   /**
-   * @brief Put object in the container with a new Index for that
+   * @brief Put the object in the container with given index.
+   *
+   * It fails when the given index is already taken or @c index is Undefined.
+   *
+   * @param[in] object Object to be pushed
+   * @param[in] index Index associated with the object
+   * @return @c index if successful, an Undefined index otherwise
+   */
+  Index push(std::unique_ptr<Object> &&object, Index index)
```

Comment:
@wateret Two push(...) work differently. `push() with Index` checks the uniqueness of `index` while `push() w/o Index` checks the validity (whether it is `UNDEFINED` or not) only. I don't think `Index` is not guaranteed not in use for latter case.

I am curious what is the use case of specifying Index? 
- Do you want to re-use some Index which is not used any longer again? Or 
- Do you want to use Index bigger than current `_next_index`? Then why?

Reply:
> while `push() w/o Index` checks the validity (whether it is `UNDEFINED` or not) only. I don't think Index is not guaranteed not in use for latter case.

We always keep `_next_index` as (highest index in the container + 1).  So it does not have to check every time. I'll add comment for that. (It is just to make sure not reusing some indices which could have been existed before)

> I am curious what is the use case of specifying Index?

I should've mentioned the background for needing this. 😅 

Here's what I'm going to do #5352 , and in the figure from https://github.com/Samsung/ONE/issues/5352#issuecomment-747361763 I'm changing the backend gets a partial subgraph(only the part for each backend needs). However all the operation/operand indices must be kept, so there should be something like this.